File: Backend/MyBackend/Models/SessionPost.cs
using MyBackend.Entities;
using System.ComponentModel.DataAnnotations;

namespace MyBackend.Models
{
    //client side validation
    public class SessionPost
    {
        public GameSystemModel? System { get; set; }

        public string Title { get; set; } = string.Empty;

        public string Tags { get; set; } = string.Empty;

        public string Triggers { get; set; } = string.Empty;

        public string Date { get; set; } = string.Empty;
        public bool IsRemote { get; set; }
        //Like city or town, locale - (napisz to na froncie)
        public string Location { get; set; } = string.Empty;
        public string Vtt { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public string Picture { get; set; } = string.Empty;
        //public IFormFile? Picture { get; set; }

        public int MaxNumberOfPlayers { get; set; }

        //logged in user, will be different in the finished app
        public string GmUserName { get; set; } = string.Empty;
    }
}

File: Frontend/src/app/models/SessionPost.ts
import { GameSystem } from "./GameSystem"
import { GmForSession } from "./GmForSession"
import ISessionPost from "./ISessionPost"

export class SessionPost implements ISessionPost{
    triggers: string = ""
    vtt: string = ""
    location: string = ""
    date: string | undefined = ""
    isRemote: boolean = false
    system : GameSystem = new GameSystem()
    title : string = ""
    tags : string = ""
    description : string = ""
    maxNumberOfPlayers : number = 0
    picture: string = ""
    //picture : FormData = new FormData()
    gmUserName : string = ""
}